File: client/scala/libs/spark.ml/src/main/scala-2.11/edu/mit/csail/db/ml/modeldb/client/event/ExperimentEvent.scala
package edu.mit.csail.db.ml.modeldb.client.event

import com.twitter.util.Await
import edu.mit.csail.db.ml.modeldb.client.ModelDbSyncer
import modeldb.ModelDBService.FutureIface

/**
  * Store experiment run on the server.
  */
case class ExperimentEvent(experiment: modeldb.Experiment) extends ModelDbEvent {
  override def sync(client: FutureIface, mdbs: Option[ModelDbSyncer]): Unit = {
    val res = Await.result(client.storeExperimentEvent(modeldb.ExperimentEvent(experiment)))
    mdbs.get.experiment = mdbs.get.experiment.copy(id = res.experimentId)
  }
}

File: client/python/modeldb/events/ExperimentEvent.py
"""
Store Experiment on server.
"""
from modeldb.events.Event import Event
from ..thrift.modeldb import ttypes as modeldb_types


class ExperimentEvent(Event):
    """
    Class for creating and storing ExperimentEvents
    """

    def __init__(self, experiment):
        self.experiment = experiment

    def make_event(self, syncer):
        """
        Constructs a thrift ExperimentEvent object with appropriate fields.
        """
        return modeldb_types.ExperimentEvent(self.experiment)

    def sync(self, syncer):
        """
        Stores ExperimentEvent on the server.
        """
        thrift_client = syncer.client
        res = thrift_client.storeExperimentEvent(self.make_event(syncer))
        syncer.experiment.id = res.experimentId
